### PR TITLE
FIx: lifecycle hook `update` should not been called after panel is closed or not ready yet

### PR DESCRIPTION
**Pull request by doommm**
@arsen2010 @VisualSJ 

Re: #

### Changelog

* lifecycle hook `update` should not been called after panel is closed or not ready yet

### Dependent PR

* https://github.com/cocos/creator-ui-kit/pull/291

### Related issue

- https://github.com/cocos/3d-tasks/issues/14251
- https://github.com/cocos/3d-tasks/issues/14230

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: editor/inspector/contributions/node.js [hunk 1]
```diff
@@ -385,13 +385,16 @@ const Elements = {
...
+            panel.__nodeChangedHandle__ = undefined;
 
             panel.__nodeChanged__ = (uuid) => {
                 if (Array.isArray(panel.uuidList) && panel.uuidList.includes(uuid)) {
-                    window.cancelAnimationFrame(animationId);
-                    animationId = window.requestAnimationFrame(async () => {
+                    window.cancelAnimationFrame(panel.__nodeChangedHandle__);
+                    panel.__nodeChangedHandle__ = window.requestAnimationFrame(async () => {
                         for (const prop in Elements) {
+                            if (!panel._ready) {
```
Comment (arsen2010) [inline editor/inspector/contributions/node.js]: 换成正式接口的方式 panel.isReady() 或者正式的对外变量 panel.ready

Comment (doommm) [reply editor/inspector/contributions/node.js]: 已调整为 `panel.ready`

File: editor/inspector/components/class.js [hunk 2]
```diff
@@ -94,6 +94,12 @@ exports.methods = {
  */
 async function update(dump) {
     const $panel = this;
+
+    if (!$panel.$this.isConnected) {
+        // console.warn('host is disconnected, should not update');
```
Comment (arsen2010) [inline editor/inspector/components/class.js]: 去掉注释

**Review by arsen2010 — APPROVED** [review]

File: editor/inspector/components/class.js (hunk 2 above)
Comment (doommm) [reply editor/inspector/components/class.js]: 已去除

**Comment by doommm** [comment]
@VisualSJ

File: editor/inspector/contributions/node.js [hunk 3]
```diff
@@ -1847,6 +1859,8 @@ exports.ready = async function ready() {
 
     // 为了避免把 ui-num-input, ui-color 的连续 change 进行 snapshot
     panel.snapshotLock = false;
+    // 标记位，ready 时设为 true，close 时设为 false。给异步操作提供一个标记位
```
Comment (VisualSJ) [inline editor/inspector/contributions/node.js]: English

**Pull request merged**